great_expectations/expectations/expectation.py

```diff
@@ -1340,36 +1340,30 @@ def _validate(
         total_count = metrics.get("table.row_count")
         null_count = metrics.get("column_values.nonnull.unexpected_count")
         unexpected_count = metrics.get(self.map_metric + ".unexpected_count")
+        unexpected_values = metrics.get(self.map_metric + ".unexpected_values")
+        unexpected_index_list = metrics.get(self.map_metric + ".unexpected_index_list")
 
-        success = None
         if total_count is None or null_count is None:
+            total_count = nonnull_count = 0
+        else:
+            nonnull_count = total_count - null_count
```

Comment:
For the `get()`'s above, can we set the default value to be `0` as opposed to `None`? This would simplify type checking and save us the effort of manually having to cast from `None -> 0`.

```
    if total_count == 0 or null_count == 0:
        nonnull_count = 0
    else:
        nonnull_count = total_count - null_count
```